app/server/api/poli_payments_api.py

```diff
@@ -0,0 +1,87 @@
+from flask import Blueprint, request, make_response, jsonify
+from flask.views import MethodView
+from server import db
+from server.utils.poli_payments import PoliPaymentsError, get_poli_link_status, get_poli_link_from_token
+from server.models.fiat_ramp import FiatRamp, FiatRampStatusEnum
+from server.models.user import User
+from server.utils.credit_transfers import make_deposit_transfer, find_user_with_transfer_account_from_identifiers
+
+poli_payments_blueprint = Blueprint('poli_payments_blueprint', __name__)
+
+
+class PoliPaymentWebhookAPI(MethodView):
+    def post(self):
+        """
+        POLi Payments Webhook that provides a token to lookup a POLi link and check status.
+
+        :return:
+        """
+
+        if request.data:
+            try:
+                response = get_poli_link_from_token(token=request.data.decode())
```

Comment:
I'm confused as to what this is doing... it's the poli webhook that then loops back to make a request from.... the original poli server? why do they need this?

Reply:
Yes, it's an authing process.

They send us a unique token to our webhook, we use that to request a Poli link, which we can request the status from.